### DISCUSSION TITLE
Manual speaker managment to remove duplicates

### DISCUSSION THREAD

**Discussion by ethanzrd** [juanmc2005/diart#171] [Q&A]
Hey,
I was wondering whether it'd be possible to manually manage speakers and check for duplicates every once in a while, as the pipeline can generate sometimes multiple speakers which end up being another existing speaker.
How would one tackle this with this library?
Thank you!

**Comment by juanmc2005** [comment]
Hi @ethanzrd, I see two options to reduce duplicate speakers:

1. Increase the value of `delta_new`, which is essentially the distance threshold (between speaker embedding and closest cluster centroid) to detect new speakers.
2. Prune centroids in `OnlineSpeakerClustering`, for example when they are too close to each other you can merge them, or when they aren't used for a long time you can erase them. My experiments on this weren't very successful in terms of DER but it may be worth it to improve readability.
